worker-message-prioritization-rerouting/src/main/java/com/github/workerframework/workermessageprioritization/rerouting/StagingQueueCreator.java

```diff
@@ -57,6 +91,13 @@ void createStagingQueue(final Queue targetQueue, final String stagingQueueName)
                     "IOException thrown creating or checking staging queue when calling channel.queueDeclare(%s, %s, %s, %s, %s)",
                     stagingQueueName, durable, exclusive, autoDelete, arguments), iOException);
 
+            try {
+                reconnectToRabbitMQ();
+            } catch (final IOException | TimeoutException ignored) {
```

Comment:
I noticed this on the channel.isOpen() JavaDoc

> Determine whether the component is currently open. Will return false if we are currently closing. Checking this method should be only for information, because of the race conditions - state can change after the call. Instead just execute and try to catch **ShutdownSignalException** and IOException

The ShutdownSignalException is a Runtime Exception

Reply:
I've added this exception to the catch of close call here: https://github.com/WorkerFramework/worker-message-prioritization/pull/12/files#diff-42c3903cee6b7168334952cf8d1d365ffbfe27774102d6b878acbe42c9eeaca2R68